tests_dags/dag_tester.py

```diff
@@ -0,0 +1,310 @@
+"""
+General entry point for testing end to end dags
```

Comment:
This is the dag testing framework.

![image](https://www.lucidchart.com/publicSegments/view/f398403a-729c-40b1-bbeb-51f6d5c3f4d7/image.png)

`AbstractEndToEndTest` is the abstract test super class that is extended by `EndToEndBackfillJobTest` and `EndToEndSchedulerJobTest`.

Actual tests classes that developers will write should extend `EndToEndBackfillJobTest` and/or `EndToEndSchedulerJobTest` depending on whether their scenario should be tested through a `BackfillJob`, a `SchedulerJob` or both.

When a class enherits from `EndToEndBackfillJobTest` or `EndToEndSchedulerJobTest`, nose automatically detects the `test_backfilljob` and `test_schedulerjob` methods and adds them to the list of tests to execute.

The `test_backfilljob` and `test_schedulerjob` methods respectively use a `BackFillJobRunner` and a `SchedulerJobRunner`. They each extend the `Runner` class which contains the test context preparation. They each then respectively run a `BackfillJob` and a `SchedulerJob` with the test context.